Create docs for the VDC dashboard package (#2866)
threefoldtech/js-sdk: docs/vdc/vdc_dashboard.md

## The VDC dashboard architecture
```
  ├── chats
  ├── scripts
  ├── bottle
  │   └── deployments.py
  │   └── backup.py
  │   └── models.py
  │   └── vdc_helpers.py
  ├── services
  │   └── domain.py
  │   └── etcd_backup.py
  │   └── k8s_auto_extend.py
  │   └── provisioning_wallet_billing.py
  │   └── s3_auto_topup.py
  │   └── transaction_from_prepain_to_provisioning.py
  │   └── zdb_auto_topup.py
  ├── frontend
  │   └── assets
  │   └── components
  │   └── index.js
  │   └── api.js
  │   └── index.html
  │   └── app.vue
  ├── package.py
  ├── package.toml
  └── __init__.py
  ```
### REST APIs
- There is a [REST Interfaces](./vdc_dashboard_rest_interface.md) that make the user do some actions to control and monitor the VDC.
- Some actions can be applied with the REST APIs like:
  - `Get all VDC info`
  - `Get all worker nodes`
  - `Add new worker node`
  - `Delete existing worker node`
  - `Get all storage nodes`
  - `Add new storage node`
  - `Delete storage node`
  - `Get vdc prepaid wallet info`
  - `Get all used pools`
  - `Get alerts`
### The Package Frontend

> - The frontend built with [Vue js](https://vuejs.org/) and use [vuetify js](https://vuetifyjs.com/) some components.

Reviewer: ```suggestion
- The frontend is built with [Vue js](https://vuejs.org/) using some components from [vuetify js](https://vuetifyjs.com/).
```

- The VDC Marketplace home page:

>   - The user can deploy, list and delete his/her solutions.

Reviewer: ```suggestion
  - The user can deploy, list, and delete their solutions.
```

![vdc_marketplace](./images/vdc_marketplace.png)

- The VDC setting home page:

>   - The default tab represent the `Compute Nodes` for the VDC Kubernetes cluster, The user can Add or remove the nodes.

Reviewer: ```suggestion
  - The default tab represents the `Compute Nodes` for the VDC Kubernetes cluster where the user can add or remove nodes from their VDC.
```

>   - The default tab represent the `Compute Nodes` for the VDC Kubernetes cluster, The user can Add or remove the nodes.

Reviewer: ```suggestion
  - `Compute Nodes tab` is the default tab for the VDC Kubernetes cluster, where the user can add or remove their VDC's nodes.
```

  ![vdc_setting](./images/vdc_setting.png)
  - In the `Storage Nodes` tab the user can:

>     - Add/Delete storage node.

Reviewer: ```suggestion
    - Add/Delete storage nodes.
```

    - Enable quantum storage.
    - Download [Z-stor](https://github.com/threefoldtech/0-stor_v2) configuration.
    - Download ZDBs information.

    ![vdc_storage_nodes](./images/vdc_storage_nodes.png)

>   - In Wallet Information tab there is the prepaid wallet information.

Reviewer: ```suggestion
  - `Wallet Information tab`: consists of the prepaid wallet information.
```

    ![vdc_storage_nodes](./images/vdc_wallet.png)

>   - In the Backup And Restore tab there is a list of backups and the user can create new and restore any of them.

Reviewer: ```suggestion
  - `Backup And Restore tab`: consists of a list of backups. It also enables the user to create new ones or restore any of the previously backed up ones.
```

# VDC Deployer

This manual will go through the `VDC_dashboard` package and how to add new solutions to its marketplace.

- The `VDC dashboard` is the end user interface, where the user can interact with their VDC Settings and Marketplace.

## Requirement
- At lest One VDC instance configured on the machine.

## The VDC dashboard architecture
```
  ├── chats
  ├── scripts
  ├── bottle
  │   └── deployments.py
  │   └── backup.py
  │   └── models.py
  │   └── vdc_helpers.py
  ├── services
  │   └── domain.py
  │   └── etcd_backup.py
  │   └── k8s_auto_extend.py
  │   └── provisioning_wallet_billing.py
  │   └── s3_auto_topup.py
  │   └── transaction_from_prepain_to_provisioning.py
  │   └── zdb_auto_topup.py
  ├── frontend
  │   └── assets
  │   └── components
  │   └── index.js
  │   └── api.js
  │   └── index.html
  │   └── app.vue
  ├── package.py
  ├── package.toml
  └── __init__.py
  ```

### Background Services

> - `domain.py`: This service is responsible for redeploy subdomain.

Reviewer: ```suggestion
- `domain.py`: This service is responsible for redeploying subdomains.
```

> - `etcd_backup.py`: This service is responsible for use `etcdctl` to get a snapshot of etcd and back it up using `restic`, *it stopped for now*

Reviewer: ```suggestion
- `etcd_backup.py`: This service is responsible for backup using `etcdctl` to get a snapshot of etcd and backing it up using `restic`. [currently disabled]
```

> - `k8s_auto_extend.py`: This service is responsible for auto extend the kubernetes cluster to its plan limits if needed.
> - `provisioning_wallet_billing.py`: This service is responsible for extend the VDC pool.
> - `s3_auto_topup.py`: *it stopped for now*
> - `transaction_from_prepain_to_provisioning.py`: This service is responsible for transfer funds from prepaid to provision wallet
> - `zdb_auto_topup.py`: This service is responsible for extend the VDC ZDBs capacity.

Reviewer: ```suggestion
- `k8s_auto_extend.py`: This service is responsible for auto extending the kubernetes cluster to its plan limits if needed.
- `provisioning_wallet_billing.py`: This service is responsible for extending the VDC pool.
- `s3_auto_topup.py`: *it stopped for now*
- `transaction_from_prepain_to_provisioning.py`: This service is responsible for transfering funds from the prepaid to the provision wallet
- `zdb_auto_topup.py`: This service is responsible for extending the VDC ZDB's capacity.
```
